File: Design-Patterns-Using-Java/Bridge/BridgePatternShape.java
import java.util.*;

// Abstraction abstract class
abstract class Shape {
    protected Colour imp;

    public Shape(Colour imp) {
        this.imp = imp;
    }

    abstract void draw();
}

// Rectangle class
class Rectangle extends Shape {
    public Rectangle(Colour imp) {
        super(imp);
    }

    @Override
    void draw() {
        System.out.println("Drawing Rectangle with colour " + imp.fill());
    }
}

// Circle class
class Circle extends Shape {
    public Circle(Colour imp) {
        super(imp);
    }

    @Override
    void draw() {
        System.out.println("Drawing Circle with colour " + imp.fill());
    }
}

// Colour abstract class
abstract class Colour {
    abstract String fill();
}

// Red class
class Red extends Colour {
    @Override
    String fill() {
        return "Red";
    }
}

// Green class
class Green extends Colour {
    @Override
    String fill() {
        return "Green";
    }
}

// Blue class
class Blue extends Colour {
    @Override
    String fill() {
        return "Blue";
    }
}

// Client code
public class BridgePatternShape {
    public static void main(String[] args) {
        Colour c1 = new Red();
        Shape abstraction = new Circle(c1);
        abstraction.draw();

        Colour c2 = new Green();
        abstraction = new Rectangle(c2);
        abstraction.draw();
    }
}

File: Design-Patterns-Using-CSharp/Bridge/BridgePatternShape.cs
using System;

// Abstraction abstract class
abstract class Shape
{
    protected Colour imp;

    public Shape(Colour imp)
    {
        this.imp = imp;
    }

    public abstract void Draw();
}

// Rectangle class
class Rectangle : Shape
{
    public Rectangle(Colour imp) : base(imp)
    {
    }

    public override void Draw()
    {
        Console.WriteLine("Drawing Rectangle with colour " + imp.Fill());
    }
}

// Circle class
class Circle : Shape
{
    public Circle(Colour imp) : base(imp)
    {
    }

    public override void Draw()
    {
        Console.WriteLine("Drawing Circle with colour " + imp.Fill());
    }
}

// Colour abstract class
abstract class Colour
{
    public abstract string Fill();
}

// Red class
class Red : Colour
{
    public override string Fill()
    {
        return "Red";
    }
}

// Green class
class Green : Colour
{
    public override string Fill()
    {
        return "Green";
    }
}

// Blue class
class Blue : Colour
{
    public override string Fill()
    {
        return "Blue";
    }
}

// Client code
class BridgePatternShape
{
    static void Main(string[] args)
    {
        Colour c1 = new Red();
        Shape abstraction = new Circle(c1);
        abstraction.Draw();

        Colour c2 = new Green();
        abstraction = new Rectangle(c2);
        abstraction.Draw();
    }
}

/*
Drawing Circle with colour Red
Drawing Rectangle with colour Green
*/